jazkarta/shop/browser/cart.py

```diff
@@ -47,10 +48,13 @@ def __call__(self):
 
         if 'submitted' in self.request.form:
             base_url = get_navigation_root_url()
+            args = ''
+            if self.request.form.get("_authenticator") is not None:
+                args = '?_authenticator=' + self.request.form.get("_authenticator")
```

Comment:
Yay! Any idea why this is a write operation? Session update? I would be not at all surprised if Plone's auto-CSRF triggered on persistent SESSION changes, but that's also awful.